pkg/kubectl/service.go

```diff
@@ -134,10 +147,14 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 		if len(portStringSlice) > 1 {
 			name = fmt.Sprintf("port-%d", i+1)
 		}
+		protocol := params["protocol"]
```

Comment:
Shouldn't this be the other way around? If I use --protocol I would expect it to override what protocol(s) the resource I am exposing has. Also you need to make sure whether a protocol has been specified with --protocol. Right now, you cannot without removing the default from the --protocol flag.